### DISCUSSION TITLE
Can SqlPage connect to multiple MSSQL databases?

### DISCUSSION THREAD

**Discussion by kevinr432** [sqlpage/SQLPage#653] [Q&A] [answered]
Hello everyone.  I am new to SQL Page, but I am loving it so far.  I have built a really nice family address book application already.    

Now I want to create a more advanced business application.    I have 15 clients with 15 different MSSQL database.  The databases are all identical in structure and I would like to have one set of SQL page code for the application.    I have only found one place to set up a database connection (in the sqlpage.yaml file in the folder sqlpage_config).

I would like to have the user sign on and then dynamically set up the MSSQL connection string to one of the 15 databases depending on the user login.

Does anyone know if this is possible?

**Comment by lovasoa** [comment, marked as answer]
Hi @kevinr432 , and welcome to SQLPage !

  A single sqlpage instance can connect only to a single database. You may be interested in the following past discussion about the same topic: https://github.com/sqlpage/SQLPage/discussions/48

  In short, you can either:
1. Launch multiple instances of sqlpage with the same root folder, or
2. [Connect](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-external-data-source-transact-sql?view=sql-server-ver16&tabs=dedicated) one sql server database to all the others.

  Since you are using [the datapage hosting service](https://beta.datapage.app/), Alexis will be happy to set it up for you, and give you a bulk discount for your 15 instances. I'll tell him to get back to you shortly.

**Reply by kevinr432** [reply]
Thank you so much for your suggestion.  I did search to see if this subject has already been discussed, but I am also new to GitHub and so I am not sure I was searching properly.
